src/pamiq_core/console/cui.py

```diff
@@ -0,0 +1,141 @@
+import argparse
+import cmd
+import json
+from typing import override
+
+import requests
+
+
+class Console(cmd.Cmd):
+    intro = 'Welcome to the AMI system console. "help" lists commands.\n'
+    prompt: str
+
+    def __init__(self, host: str, port: int) -> None:
+        super().__init__()
+        self._host = host
+        self._port = port
+        self._fetch_status()
+
+    @override
+    def onecmd(self, line: str) -> bool:
+        # Check command depend on WebAPI
+        cmd_name, _, _ = self.parseline(line)
+        if cmd_name in ["pause", "p", "resume", "r", "ckpt", "c", "shutdown", "s"]:
+            # Check if WebAPI available.
+            if self._fetch_status() == "offline":
+                print(f'Command "{cmd_name}" not executed. Can\'t connect AMI system.')
+                return False
+        # Perform command
+        return super().onecmd(line)
+
+    @override
+    def do_help(self, arg: str) -> None:
+        """Show all commands and details."""
+        # Store methods with the same docstring
+        # groups = {"Doscstring": ["command", "c"]}
+        groups: dict[str, list[str]] = {}
+        for attr in dir(self):
+            if attr.startswith("do_"):
+                cmd_name = attr[3:]
+                method = getattr(self, attr)
+                doc = method.__doc__ or ""
+                doc = doc
+                if doc not in groups:
+                    groups[doc] = []
+                groups[doc].append(cmd_name)
+        # Define showing order
+        _doc_order = [
+            "Show all commands and details.",
+            "Pause the AMI system.",
+            "Resume the AMI system.",
+            "Save a checkpoint.",
+            "Shutdown the AMI system.",
+            "Exit the console.",
+        ]
+        assert set(_doc_order) == set(groups.keys())
+        # Show details in the format f"{c}/{command} {Docstring}"
+        for doc in _doc_order:
+            cmds = "/".join(sorted(groups[doc], key=len))
+            print(f"{cmds:<11} {doc}")
+
+    def do_h(self, arg: str) -> None:
+        """Show all commands and details."""
+        self.do_help(arg)
+
+    def do_pause(self, arg: str) -> None:
+        """Pause the AMI system."""
+        response = requests.post(f"http://{self._host}:{self._port}/api/pause")
+        print(json.loads(response.text)["result"])
+
+    def do_p(self, arg: str) -> None:
+        """Pause the AMI system."""
+        return self.do_pause(arg)
+
+    def do_resume(self, arg: str) -> None:
+        """Resume the AMI system."""
+        response = requests.post(f"http://{self._host}:{self._port}/api/resume")
+        print(json.loads(response.text)["result"])
+
+    def do_r(self, arg: str) -> None:
+        """Resume the AMI system."""
+        return self.do_resume(arg)
+
+    def do_shutdown(self, arg: str) -> bool:
+        """Shutdown the AMI system."""
+        confirm = input("Confirm AMI system shutdown? (y/n): ")
```

Comment:
デフォルトでNoであることが伝わると良いと思う。
```suggestion
        confirm = input("Confirm AMI system shutdown? (y/[N]): ")
```